edit/source-editor.js

```diff
@@ -266,47 +278,125 @@ function createSourceEditor(style) {
+          }
+        });
+        typeInput.value = apply.type.text;
+        let timer;
+        const valueInput = $element({
+          tag: 'input',
+          className: 'applies-value',
+          value: apply.value.text,
+          oninput(e) {
+            clearTimeout(timer);
+            timer = setTimeout(applyChange, THROTTLE_DELAY, apply.value, e.target.value);
+          }
+        });
+        const removeButton = $element({
+          tag: 'button',
+          type: 'button',
+          className: 'applies-to-remove',
+          textContent: t('appliesRemove'),
+          onclick(e) {
+            const i = applies.indexOf(apply);
+            let repl;
+            let from;
+            let to;
+            if (applies.length < 2) {
+              alert('Can\'t remove last applies-to');
+              return;
+            }
+            if (i === 0) {
+              from = apply.mark.find().from;
+              to = applies[i + 1].mark.find().from;
+              repl = '';
+            } else if (i === applies.length - 1) {
+              from = applies[i - 1].mark.find().to;
+              to = apply.mark.find().to;
+              repl = '';
+            } else {
+              from = applies[i - 1].mark.find().to;
+              to = applies[i + 1].mark.find().from;
+              repl = ', ';
+            }
+            cm.replaceRange(repl, from, to, 'appliesTo');
+            clearApply(apply);
+            e.target.closest('li').remove();
+            applies.splice(i, 1);
+          }
+        });
+        const addButton = $element({
+          tag: 'button',
+          type: 'button',
+          className: 'applies-to-add',
+          textContent: t('appliesAdd'),
+          onclick(e) {
+            const i = applies.indexOf(apply);
+            const pos = apply.mark.find().to;
+            const text = `, ${apply.type.text}("")`;
+            cm.replaceRange(text, pos, pos, 'appliesTo');
+            const index = cm.indexFromPos(pos);
+            const newApply = {
+              type: {
+                text: apply.type.text
+              },
+              value: {
+                text: ''
+              }
+            };
+            newApply.start = index + 2;
+            newApply.type.start = newApply.start;
+            newApply.type.end = newApply.type.start + newApply.type.text.length;
+            newApply.value.start = newApply.type.end + 2;
+            newApply.value.end = newApply.value.start + newApply.value.text.length;
+            newApply.end = newApply.value.end + 2;
+            setupApplyMarkers(newApply);
+            applies.splice(i + 1, 0, newApply);
+            const li = e.target.closest('li');
+            li.parentNode.insertBefore($element({
+              tag: 'li',
+              appendChild: makeInput(newApply)
+            }), li.nextSibling);
+          }
+        });
+        return [typeInput, valueInput, removeButton, addButton];
```

Comment:
Should we also include a "RegExp test" button when the "URLs matching the regexp" option is set?